pkg/issuer/acme/dns/dns.go

```diff
@@ -330,25 +330,29 @@ func (s *Solver) solverForChallenge(ctx context.Context, issuer v1alpha2.Generic
 		}
 	case providerConfig.AzureDNS != nil:
 		dbg.Info("preparing to create AzureDNS provider")
-		clientSecret, err := s.secretLister.Secrets(resourceNamespace).Get(providerConfig.AzureDNS.ClientSecret.Name)
-		if err != nil {
-			return nil, nil, fmt.Errorf("error getting azuredns client secret: %s", err)
-		}
+		secret := ""
+		if providerConfig.AzureDNS.ClientID != "" {
```

Comment:
Add a comment here saying that if `clientID` is *empty*, managed identity will attempt to be used (or words to that effect) - it'll make it far easier for the next person touching this area of code! 😄